kvirt/providers/hcloud/__init__.py

```diff
@@ -116,21 +110,44 @@ def create(self, name, virttype=None, profile='', flavor=None, plan='kvirt', cpu
 
             placement_group = response.placement_group
 
-        created_vm = self.conn.servers.create(
-            name=name,
-            server_type=servertype,
-            image=hetzner_image,
-            start_after_create=False,
-            user_data=userdata,
-            volumes=[],
-            ssh_keys=hetzner_ssh_keys,
-            location=self.location,
-            public_net=ServerCreatePublicNetwork(enable_ipv4=False, enable_ipv6=False),
-            labels=labels,
-            placement_group=placement_group
-        )
-
-        created_vm.action.wait_until_finished(300)
+        if flavor is not None or "flavor_options" not in overrides:
```

Comment:
you should change this section because flavor_options will be undefined if flavor is None and flavor_options is not in overrides

Reply:
I can add a hard error if you have defined neither, but before the call to hcloud would also fail if you had no flavor defined as that would lead to flavor = None